plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -615,11 +451,10 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
                 .toArray(Type[]::new);
 
         AppendFiles appendFiles = transaction.newFastAppend();
+        FileIO fileIO = transaction.table().io();
         for (CommitTaskData task : commitTasks) {
-            HdfsContext context = new HdfsContext(session);
-
             DataFiles.Builder builder = DataFiles.builder(icebergTable.spec())
-                    .withInputFile(new HdfsInputFile(new Path(task.getPath()), hdfsEnvironment, context))
+                    .withInputFile(fileIO.newInputFile(task.getPath()))
```

Comment:
why this change? What about logic which was a part of `HdfsInputFile` (it contains workaround for https://github.com/trinodb/trino/issues/5201, this I guess should not be needed here, but please double check that we are not missing something).

Also it seems unrelated change, hence it should be separate preparatory commit.

Reply:
`fileIO.newInputFile` calls exactly `new HdfsInputFile(new Path(task.getPath()), hdfsEnvironment, context)`. This change is to avoid introducing `hdfsEnvironment` and `hdfsContext` to the `IcebergMetadata`.

Reply:
I think this is a rebase conflict. We changed to `withPath()` and `withFileSizeInBytes()` in #8572 to avoid an extra stat call on the coordinator.